Source/FakeItEasy.Tests/Creation/DefaultFakeCreatorTests.cs

```diff
@@ -114,6 +121,21 @@ public void CreateFake_should_pass_options_builder_that_returns_itself_for_any_c
             Assert.That(call.Invoke(builderPassedToAction), Is.SameAs(builderPassedToAction));
         }
 
+        [TestCaseSource("invalidOptionBuilderCalls")]
+        public void CreateFake_should_throw_when_type_passed_to_options_builder_is_not_an_interface(Func<IFakeOptionsBuilder<Foo>, IFakeOptionsBuilder<Foo>> call)

```

Comment:
For `CreateFake_should_pass_options_builder_that_returns_itself_for_any_call`, the `Wrapping.Implements` case was just added because `Wrapping` _doesn't actually return itself_—it's an oddball, but putting another action on the end of `Wrapping` means we do get the original object back again. I picked `Implements` out of a hat for the additional action.
So I think the first test case in `invalidOptionBuilderCalls` could be removed. It's not wrong or harmful, but may be superfluous.

More confusing to me is how the third case is related to `CreateFake_should_throw_when_type_passed_to_options_builder_is_not_an_interface`.
`WithArgumentsForConstructor` seems completely unrelated to passing interfaces. Sure, `() => CreateFoo()` is not a good value, but we wouldn't require an interface there: the problem is that we've been passed something that's not a constructor call.
It's a good test case, but I think it's been lumped into another test with a different purpose. I think the list of test cases with a more generic name or (my preference) completely separate tests with specific names would make more sense.
While I'm on the subject of names, I notice now that the existing test name is probably a little too generic. If we go with specific test names, how do you feel about `CreateFake_should_throw_when_type_passed_to_implements_is_not_an_interface`? It more closely matches the existing `CreateFake_should_pass_all_interfaces_from_implements_to_fake_and_dummy_manager`. Also, `Implements` is the only thing that cares about interfaces.